rio_tiler/io/stac.py

```diff
@@ -181,200 +186,28 @@ def __exit__(self, *args):
-            data = apply_expression(blocks, assets, data)
-
-        return data, mask
-
-    def point(
-        self,
-        lon: float,
-        lat: float,
-        assets: Union[Sequence[str], str] = None,
-        expression: Optional[str] = "",  # Expression based on asset names
-        asset_expression: Optional[
-            str
-        ] = "",  # Expression for each asset based on index names
-        **kwargs: Any,
-    ) -> List:
-        """Read a value from COGs."""
-        if isinstance(assets, str):
-            assets = (assets,)
-
-        if expression:
-            assets = self._parse_expression(expression)
-
-        if not assets:
-            raise MissingAssets(
-                "assets must be passed either via expression or assets options."
-            )
-
-        asset_urls = self._get_href(assets)
-        point = multi_point(asset_urls, lon, lat, expression=asset_expression, **kwargs)
-
-        if expression:
-            blocks = expression.split(",")
-            point = apply_expression(blocks, assets, point).tolist()
-
-        return point
-
-    def stats(
-        self,
-        pmin: float = 2.0,
-        pmax: float = 98.0,
-        hist_options: Optional[Dict] = None,
-        assets: Union[Sequence[str], str] = None,
-        **kwargs: Any,
-    ) -> Dict:
-        """Return array statistics from COGs."""
-        if not assets:
-            raise MissingAssets("Missing 'assets'")
-
-        if isinstance(assets, str):
-            assets = (assets,)
-
-        asset_urls = self._get_href(assets)
-
-        stats = multi_stats(asset_urls, pmin, pmax, hist_options=hist_options, **kwargs)
-        return {asset: stats[ix] for ix, asset in enumerate(assets)}
-
-    def info(self, assets: Union[Sequence[str], str] = None) -> Dict:
-        """Return info from COGs."""
-        if not assets:
-            raise MissingAssets("Missing 'assets'")
-
-        if isinstance(assets, str):
-            assets = (assets,)
-
-        asset_urls = self._get_href(assets)
-
-        infos = multi_info(asset_urls)
-        return {asset: infos[ix] for ix, asset in enumerate(assets)}
-
-    def metadata(
-        self,
-        pmin: float = 2.0,
-        pmax: float = 98.0,
-        assets: Union[Sequence[str], str] = None,
-        **kwargs: Any,
-    ) -> Dict:
-        """Return array statistics from COGs."""
-        if not assets:
-            raise MissingAssets("Missing 'assets'")
-
-        if isinstance(assets, str):
-            assets = (assets,)
-
-        asset_urls = self._get_href(assets)
-
-        stats = multi_metadata(asset_urls, pmin, pmax, **kwargs)
-        return {asset: stats[ix] for ix, asset in enumerate(assets)}
+    @property
+    def spatial_info(self) -> Dict:
+        """Return Dataset's spatial info."""
+        return {
+            "bounds": self.bounds,
+            "center": self.center,
+            "minzoom": self.minzoom,
+            "maxzoom": self.maxzoom,
+        }
```

Comment:
Before we were creating a Class with COGReader as parent so both `center` and `spatial_info` existed. I added those back for compatibility